src/eventalign.c

```diff
@@ -1712,18 +1717,36 @@ void realign_read(char* ref,
         //     emit_event_alignment_tsv(tsv_fp, sr, strand_idx, params, alignment);
         // }
 
+        // if(summary_fp != NULL)
+        //     fprintf(stderr, "debug summary_fp != NULL alignment.size() = %d \n",alignment.size());
+        
+        // if(summary.num_events > 0)
+        //     fprintf(stderr, "debug summary.num_events > 0 read_idx = %d \n",read_idx);
+        
         if(summary_fp != NULL && summary.num_events > 0) {
-            //assert(params.alphabet == "");
-            //const PoreModel* pore_model = params.get_model();
-            //SquiggleScalings& scalings = sr.scalings[strand_idx];
+            // write(append) f5c eventalign summary
+            std::ofstream f5c_event_align("test/ecoli_2kb_region/f5c_event_align.summary",std::ofstream::out | std::ofstream::app);
             int strand_idx = 0;
-             fprintf(summary_fp, "%zu\t%s\t", read_idx, read_name.c_str());
-             fprintf(summary_fp, "%s\t%s\t", "dna", strand_idx == 0 ? "template" : "complement");
-            fprintf(summary_fp, "%d\t%d\t%d\t%d\t", summary.num_events, summary.num_steps, summary.num_skips, summary.num_stays);
-            fprintf(summary_fp, "%.2lf\t%.3lf\t%.3lf\t%.3lf\t%.3lf\n", summary.sum_duration/(4000.0), scalings.shift, scalings.scale, 0.0, scalings.var);
+            f5c_event_align << read_idx << "\t";
+            f5c_event_align << read_name.c_str() << "\t";
+            // f5c_event_align << "dna" << "\t";
+            f5c_event_align << "template" << "\t";
+            f5c_event_align << summary.num_events << "\t";
+            f5c_event_align << summary.num_steps << "\t";
+            f5c_event_align << summary.num_skips << "\t";
+            f5c_event_align << summary.num_stays << "\t";
+            f5c_event_align << summary.sum_duration/(4000.0) << "\t";
+            f5c_event_align << scalings.shift << "\t";
+            f5c_event_align << scalings.scale << "\t";
+            f5c_event_align << 0.0 << "\t";
+            f5c_event_align << scalings.var << "\n";
+            //  fprintf(summary_fp, "%zu\t%s\t", read_idx, read_name.c_str());
+            //  fprintf(summary_fp, "%s\t%s\t", "dna", strand_idx == 0 ? "template" : "complement");
+            // fprintf(summary_fp, "%d\t%d\t%d\t%d\t", summary.num_events, summary.num_steps, summary.num_skips, summary.num_stays);
+            // fprintf(summary_fp, "%.2lf\t%.3lf\t%.3lf\t%.3lf\t%.3lf\n", summary.sum_duration/(4000.0), scalings.shift, scalings.scale, 0.0, scalings.var);
+
```

Comment:
could you send a pull request without these C++ likefile writing changes? I would like to keep fprintf

you can simply keep the previous fprintfs and change
FILE* summary_fp = stderr; to
FILE * summary_fp = fopen(test/ecoli_2kb_region/f5c_event_align.summary,"w");
write to a file
and at the end close the file by
fclose(summary_fp);

Reply:
done. test_eventalign.awk may have to be changed. that depends on the nanopolish output file. f5c does not store the fast5_path, hence deleted the fast5_path col from nanopolish file.